test.py

```diff
@@ -809,6 +827,41 @@ def write_junit_failure_report(self, xml_res: ET.Element) -> None:
             system_err.text = read_log(self.server_log_filename)
 
 
+class TopologyTest(PythonTest):
+    """Run a pytest collection of cases against Scylla clusters handling topology changes"""
+    is_before_test_ok: bool
+    is_after_test_ok: bool
+    status: bool
+
+    def __init__(self, test_no: int, shortname: str, suite) -> None:
+        super().__init__(test_no, shortname, suite)
+
+    async def run(self, options: argparse.Namespace) -> Test:
+
+        async with get_cluster_manager(self.shortname, self.suite.clusters) as manager:
+            self.args.insert(0, "--host={}".format(manager.cluster[0].host))
```

Comment:
Nit: a local `cluster = manager.cluster` variable would make the code more readable.

Reply:
The cluster can change inside manager. For topology tests this will be more often than not. The exception handling code later could have a dangling cluster reference and would bring irrelevant logs.

Reply:
I know that the cluster object can mutate. But can `manager.cluster` really change, i.e. point to one cluster first, then another cluster - during the same test? How could this happen? And if this can happen - won't we have some big problems? Because that would bean we could call `before_test` on a different object than `after_test` and so on.